include/glow/Base/Tensor.h

```diff
@@ -1045,6 +1045,53 @@ inline size_t getFlattenedOffset(llvm::ArrayRef<dim_t> strides,
   return index;
 }
 
+/// Helper function which \returns true if a slice with the shape \p sliceShape
+/// references fully a tensor with the shape \p tensorShape. This happens when
+/// the shapes are the same.
+inline bool isSliceFull(llvm::ArrayRef<dim_t> sliceShape,
+                        llvm::ArrayRef<dim_t> tensorShape) {
+  assert(sliceShape.size() == tensorShape.size() &&
+         "Array length mismatch for slice/tensor sizes!");
+  for (size_t dim = 0, dimEnd = sliceShape.size(); dim < dimEnd; ++dim) {
+    if (sliceShape[dim] != tensorShape[dim]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/// Helper function which \returns true if a slice with the shape \p sliceShape
+/// referenced from a larger tensor with the shape \p tensorShape is contiguous
+/// in memory (assuming the tensor it is referenced from is contiguous). This
+/// happens when the slice dimensions:
+/// - Start with singleton dimensions (dimensions equal to 1).
+/// - Continue with a partially extracted dimension (one maximum).
+/// - End with fully extracted dimensions.
+inline bool isSliceContiguous(llvm::ArrayRef<dim_t> sliceShape,
```

Comment:
nit: Please move the body of this function into `Tensor.cpp`.

Reply:
Done.